src/core/src/enclave_sgx.rs

```diff
@@ -447,3 +447,90 @@ fn check_verification_report(
     // AVR verification done
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use enclave_sgx::*;
+
+    #[test]
+    fn test_init_enclave() {
+        let mut enclave = EnclaveConfig::default();
+
+        enclave.initialize_enclave();
+        assert_eq!( (enclave.enclave_id.handle > 0), true);
+    }
+
+    #[test]
+    fn test_get_epid_group() {
+        let mut enclave = EnclaveConfig::default();
+        enclave.initialize_enclave();
+        assert_eq!( (enclave.enclave_id.handle > 0), true);
+
+        let ret = enclave.get_epid_group();
+        assert_eq!((ret.len() > 0), true);
+    }
+
+    #[test]
+    fn test_is_sgx_simulator() {
+        let mut enclave = EnclaveConfig::default();
+        enclave.initialize_enclave();
+        assert_eq!( (enclave.enclave_id.handle > 0), true);
+
+        let ret = enclave.check_if_sgx_simulator();
+        println!("is_sgx_simulator ? {:?}", ret);
```

Comment:
check if this function is returning mode same as env variable

Reply:
Ok